src/tasks/tasks.ts

```diff
@@ -125,83 +193,81 @@ export function disableStackCapturing() {
 }
 
 /*
- * map
+ * Operators (map, bimap, all)
  */
-export function map({type, payload, success, error}: Task, transform: Function): Task {
-  return {
-    type,
-    payload,
-    success: (...args) => success(transform(...args)),
-    error
-  };
-}
 
-export function bimap(
-  {type, payload, success, error}: Task,
-  successTransform: Function,
-  errorTransform: Function
-) {
-  return {
-    type,
-    payload,
-    success: (...args) => success(successTransform(...args)),
-    error: (...args) => error(errorTransform(...args))
-  };
+// This abomination is because TypeScript does not have
+// higher kinded types.
+interface TaskExport {
+  all<TAll>(tasks: TAll[]): Task<any, TAll[]>;
+
+  all<P1, T1>(tasks: [Task<P1, T1>]): Task<any, [T1]>;
+
+  all<P1, T1, P2, T2>(
+    tasks: [Task<P1, T1>, Task<P2, T2>]
+  ): Task<any, [T1, T2]>;
+
+  all<P1, T1, P2, T2, P3, T3>(
+    tasks: [Task<P1, T1>, Task<P2, T2>, Task<P3, T3>]
+  ): Task<any, [T1, T2, T3]>;
+
+  map<P, T, R>(t: Task<P, T>, f: Transformer<T, R>): Task<P, R>;
+  bimap<P, T, R>(t: Task<P, T>, f: Transformer<T, R>, f2: Function): Task<P, R>;
 }
 
-// This is kind of weird; handlers require that the task itself is
-// fed back in to the
-const compositeHandler = tasks => ({success, error}) => {
-  if (tasks.length === 0) {
-    return success([]);
-  }
-  let unsettled = tasks.length;
-  let failed = false;
-  const results = tasks.map(_ => null);
-  return Promise.all(tasks.map((task, index) => {
-    const handler = task.type[TASK_TYPE_TO_HANDLER];
-    return handler({
-      ...task,
-      success: (value) => {
+const all = (tasks) => {
+  return _task(null, (_, success, error) => {
+    if (tasks.length === 0) {
+      return success([]);
+    }
+    const results = tasks.map(_ => null);
+    let remaining = tasks.length;
```

Comment:
What do you think about using generators for this? Never did, but feel it might be a nice play to do so ;)